### PR TITLE
scan: wire up all datasource

### LINKED ISSUE #684: Wire up all data sources page

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #684

### CODE REVIEW

File: scan/src/rpc/DataSourceHook.re [hunk 1]
```diff
@@ -13,6 +13,7 @@ module DataSource = {
     description: string,
     fee: list(TxHook.Coin.t),
     executable: JsBuffer.t,
+    timestamp: MomentRe.Moment.t,
```
Comment (taobun) [inline scan/src/rpc/DataSourceHook.re:16]: Can we use latest revision as a timestamp?

File: scan/src/rpc/TxHook.re [hunk 2]
```diff
@@ -42,6 +42,14 @@ module Coin = {
   let newCoin = (denom, amount) => {denom, amount};
 
   let getDescription = coin => (coin.amount |> Format.fPretty) ++ " " ++ coin.denom;
+
+  let getAmount = coins => {
+    let coinOpt = coins->Belt_List.get(0);
```
Comment (taobun) [inline scan/src/rpc/TxHook.re:70]: Assume that it has only one element?

Comment (evilpeach) [reply scan/src/rpc/TxHook.re:70]: Its name is misleading. It should be named `getFeeAmount`. Currently, I think we don't need to support multiple fees.

File: scan/src/pages/DataSourceHomePage.re [hunk 3]
```diff
@@ -23,10 +23,23 @@ module Styles = {
...
+      height(`px(200)),
+      boxShadow(Shadow.box(~x=`px(0), ~y=`px(2), ~blur=`px(2), Css.rgba(0, 0, 0, 0.05))),
+      backgroundColor(white),
+    ]);
 };
 
 [@react.component]
 let make = () => {
+  let dataSourcesOpt = DataSourceHook.getList();
+  Js.Console.log(dataSourcesOpt);
```
Comment (taobun) [inline scan/src/pages/DataSourceHomePage.re]: Remove log

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/pages/DataSourceHomePage.re [hunk 4]
```diff
@@ -106,57 +122,32 @@ let make = () => {
...
+               <div className=Styles.fullWidth>
+                 <Row>
+                   <Col> <HSpacing size=Spacing.xl /> </Col>
+                   <Col size=0.5>
+                     <TElement elementType={TElement.DataSource(ID.DataSource.ID(id), name)} />
+                   </Col>
+                   <Col size=0.5> <TElement elementType={timestamp->TElement.Timestamp} /> </Col>
+                   <Col size=1.> <TElement elementType={owner->TElement.Address} /> </Col>
+                   <Col size=0.4>
+                     <TElement elementType={fee->TxHook.Coin.getFeeAmount->TElement.Fee} />
```
Comment (taobun) [inline scan/src/pages/DataSourceHomePage.re:139]: Expect to reuse Fee type? or float type?

Comment (taobun) [reply scan/src/pages/DataSourceHomePage.re:139]: image

File: scan/src/rpc/DataSourceHook.re (hunk 1 above)
Comment (evilpeach) [reply scan/src/rpc/DataSourceHook.re:16]: I think we should get it from the end point.

**Review by taobun — APPROVED** [review]

**Review by prin-r — APPROVED** [review]

**Pull request merged**